csharp/src/Ice/IceSSL/SSLEngine.cs

```diff
@@ -260,12 +228,68 @@ internal void verifyPeer(ConnectionInfo info, string description)
         }
     }
 
+    internal SslClientAuthenticationOptions clientAuthenticationOptions(
+        RemoteCertificateValidationCallback remoteCertificateValidationCallback,
+        string host)
+    {
+        return new SslClientAuthenticationOptions
+        {
+            ClientCertificates = _certs,
+            LocalCertificateSelectionCallback = (sender, targetHost, certs, remoteCertificate, acceptableIssuers) =>
+            {
+                if (certs == null || certs.Count == 0)
+                {
+                    return null;
+                }
+                else if (certs.Count == 1)
+                {
+                    return certs[0];
+                }
+
+                // Use the first certificate that match the acceptable issuers.
+                if (acceptableIssuers != null && acceptableIssuers.Length > 0)
+                {
+                    foreach (X509Certificate certificate in certs)
+                    {
+                        if (Array.IndexOf(acceptableIssuers, certificate.Issuer) != -1)
+                        {
+                            return certificate;
+                        }
+                    }
+                }
+                return certs[0];
+            },
+            RemoteCertificateValidationCallback = remoteCertificateValidationCallback,
+            TargetHost = host,
+            CertificateRevocationCheckMode = X509RevocationMode.NoCheck
```

Comment:
CertificateRevocationCheckMode was previously enabled depending on the value of `IceSSL.CheckCRL` property. This would be only possible with the programmatic API.